From 293650d33069276446b286ad856cfb9854ea83e4 Mon Sep 17 00:00:00 2001
From: Richard Levitte <levitte@openssl.org>
Date: Sun, 28 Jul 2024 10:47:08 +0200
Subject: [PATCH] fix: util/mkinstallvars.pl mistreated LDLIBS on Unix (and
 Windows)

Don't do comma separation on those platforms.

Fixes #24986

Reviewed-by: Neil Horman <nhorman@openssl.org>
Reviewed-by: Matt Caswell <matt@openssl.org>
Reviewed-by: Tom Cosgrove <tom.cosgrove@arm.com>
(Merged from https://github.com/openssl/openssl/pull/25018)

(cherry picked from commit 0beef0ba00f7864b7367899d859509a99237fcf0)
---
 util/mkinstallvars.pl | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/util/mkinstallvars.pl b/util/mkinstallvars.pl
index 5fadb708e1..e2b7d9d083 100644
--- a/util/mkinstallvars.pl
+++ b/util/mkinstallvars.pl
@@ -124,7 +124,9 @@ print <<_____;
 our \$VERSION                    = '$ENV{VERSION}';
 our \@LDLIBS                     =
     # Unix and Windows use space separation, VMS uses comma separation
-    split(/ +| *, */, '$ENV{LDLIBS}');
+    \$^O eq 'VMS'
+    ? split(/ *, */, '$ENV{LDLIBS}')
+    : split(/ +/, '$ENV{LDLIBS}');
 
 1;
 _____
-- 
2.39.2

